contrib/src/main/java/com/datatorrent/contrib/avro/AvroFileInputOperator.java

```diff
@@ -0,0 +1,164 @@
+ * 
+ * @displayName AvroFileInputOperator
+ * @category Input
+ * @tags fs, file,avro, input operator
+ */
+@InterfaceStability.Evolving
+public class AvroFileInputOperator extends AbstractFileInputOperator<GenericRecord>
+{
+
+  private transient long offset = 0L;
+
+  @AutoMetric
+  int recordCnt = 0;
+
+  @AutoMetric
+  int errorCnt = 0;
+
+  private transient DataFileStream<GenericRecord> avroDataStream;
+  private transient GenericRecord record = null;
+
+  public final transient DefaultOutputPort<GenericRecord> output = new DefaultOutputPort<GenericRecord>();
+
+  public final transient DefaultOutputPort<String> completedFilesPort = new DefaultOutputPort<String>();
+
+  public final transient DefaultOutputPort<String> errorRecordsPort = new DefaultOutputPort<String>();
+
+  /**
+   * Returns a input stream given a file path
+   * 
+   * @param path
+   * @return InputStream
+   * @throws IOException
+   */
+  @Override
+  protected InputStream openFile(Path path) throws IOException
+  {
+    InputStream is = super.openFile(path);
+    if (is != null) {
+      try {
+        DatumReader<GenericRecord> datumReader = new GenericDatumReader<GenericRecord>();
+        avroDataStream = new DataFileStream<GenericRecord>(is, datumReader);
+        datumReader.setSchema(avroDataStream.getSchema());
+      } catch (NullPointerException npe) {
+        LOG.error("Schemaless file", npe);
+        throw new NullPointerException();
+      }
+    }
+    return is;
+  }
+
+  @Override
+  protected GenericRecord readEntity() throws IOException
+  {
+    return readRecord();
+  }
+
+  /**
+   * Reads a GenericRecord from the given input stream<br>
+   * Emits the FileName,Offset,Exception on the error port if its connected
+   * 
+   * @return GenericRecord
+   */
+  private GenericRecord readRecord() throws IOException
+  {
+    record = null;
+
+    try {
+      if (avroDataStream != null && avroDataStream.hasNext()) {
+        offset++;
+
+        record = avroDataStream.next();
+        recordCnt++;
+        return record;
+      }
+    } catch (AvroRuntimeException are) {
+      LOG.error("Exception in parsing record for file - " + super.currentFile + " at offset - " + offset, are);
+      if (errorRecordsPort.isConnected()) {
+        errorRecordsPort.emit("FileName:" + super.currentFile + ", Offset:" + offset);
+      }
+      errorCnt++;
+      throw new AvroRuntimeException(are);
+    }
+    return record;
+  }
+
+  @Override
+  protected void closeFile(InputStream is) throws IOException
+  {
+    String fileName = super.currentFile;
+    super.closeFile(is);
+    if (avroDataStream != null) {
+      avroDataStream.close();
+    }
+    if (completedFilesPort.isConnected()) {
+      completedFilesPort.emit(fileName);
+    }
+    offset = 0;
+  }
+
+  @Override
+  protected void emit(GenericRecord tuple)
+  {
+    if (tuple != null) {
+      output.emit(tuple);
+    }
+  }
+
+  @Override
+  public void endWindow()
+  {
+    errorCnt = 0;
```

Comment:
Noticed that autometric variables are not marked private in several operators.see CsvParser, JsonParser. They will not be accessible in the test class unless getters are provided.
Has the usage changed since they were first implemented ?
beginWindow seems a better place for resetting the values, moving it there.